src/Auth/Auth.js

```diff
@@ -68,28 +78,19 @@ export default class Auth {
   }
 
   logout = () => {
-    localStorage.removeItem("access_token");
-    localStorage.removeItem("id_token");
```

Comment:
`expires_at` should be removed from `localStorage` as well. This information is related to the tokens generated and, as such, will be useless when you reload the page. The `checkSession` method will get you a new one.

Reply:
I was using expires_at to avoid the overhead of calling renewToken when the user's session has expired or doesn't exist. Note that [I'm checking in App.js to suppress the call to renewToken](https://github.com/coryhouse/react-auth0-final/pull/1/files#diff-14b1e33d5bf5649597cdc0e4f684daddR25).

So it's a perf enhancement. Make sense?

Reply:
I see. Well, the problem here is that `expires_at` doesn't represent when the users' session at Auth0 will expire, just when the token you got will expire. Users' sessions at Auth0 last for 24h+ (by default, if I'm not mistaken), tokens usually last for a couple of hours.

The guidance on how to approach this was defined a few days ago to be honest, so we don't have any public content mentioning it (I think). But, basically, fixing this is just as simple as adding a flag (e.g., `apparentlySignedIn` < terrible var name 😄) that, when your app loads, it checks if this flag is `true` and then tries to `checkSession`. If this flag is `falsy`, the app avoids the overhead.

Reply:
What is `apparentlySignedIn` set to? I assume the value stored in localStorage?

Reply:
yes, you can persist this in `localStorage`. Initially, when your users load your app for the first time, there will be nothing there (falsy). Then, when they sign in and get redirected back from Auth0, your app will changed to `true`.

With that in place, if your users refresh (or close & reopen) your app, your app will see `true` there and will check if the session at Auth0 is still up. If it is, you will get back tokens, as you already know.

Now, when your users hit logout, your app will set this flag to false so when the app refreshes, no `checkSession` call is made.

To give more context, another alternative would be making your app's `signOut` kill the session at Auth0. But then you would kill a session that might be used for SSO from a particular app, which is not ideal.

Reply:
Thanks Bruno! I've updated my approach to store "checkSession" in memory as you suggested. Look good? 

I've also moved expires_at to an instance var.

Assuming this all looks good to you, my only reservation is how to handle the API calls. I think I'll move them back outside Auth.js to the React components, and accept that they'll make a call to this.props.auth.getAccessToken. I'm not sure moving them inside Auth.js is buying me anything.

Reply:
@brunokrebs - I just added a short 4 min clip explaining approaches for storing tokens. I'd love your feedback: https://www.dropbox.com/s/5ntaf4acpnpc4ec/react-auth0-authentication-security-m4-05.mp4?dl=0

Reply:
Yeah, I agree that moving them inside `Auth.js` doesn't really add any value.  Oh, and I just checked the clip and it looks just fine for me. I liked the comparison table you provided and that you briefly explained some trade offs.

Reply:
Excellent. Thanks so much for all your valuable input Bruno! 🔥